internal/terraform/context_plan_actions_test.go

```diff
@@ -2781,6 +2781,252 @@ resource "test_object" "a" {
 				}
 			},
 		},
+		"action expansion with unknown instances": {
+			module: map[string]string{
+				"main.tf": `
+variable "each" {
+  type = set(string)
+}
+action "test_unlinked" "hello" {
+  for_each = var.each
+}
+resource "test_object" "a" {
+  lifecycle {
+    action_trigger {
+      events = [before_create]
+      actions = [action.test_unlinked.hello["a"]]
+    }
+  }
+}
+`,
+			},
+			expectPlanActionCalled: false,
+			planOpts: &PlanOpts{
+				Mode:            plans.NormalMode,
+				DeferralAllowed: true,
+				SetVariables: InputValues{
+					"each": &InputValue{
+						Value:      cty.UnknownVal(cty.Set(cty.String)),
+						SourceType: ValueFromCLIArg,
+					},
+				},
+			},
+			assertPlan: func(t *testing.T, p *plans.Plan) {
```

Comment:
Can you check here that `test_object.a` is also deferred? I think an action being deferred should result in the resource that triggers it also being deferred? What do you think?